src/fields/field_value_cache.hh

```diff
@@ -197,30 +220,40 @@ public:
 
     /// Return begin position of element chunk in FieldValueCache
     inline unsigned int element_chunk_begin(unsigned int mesh_elm_idx) const {
-        std::map<unsigned int, unsigned int>::const_iterator it = element_to_map_.find(mesh_elm_idx);
+        std::unordered_map<unsigned int, unsigned int>::const_iterator it = element_to_map_.find(mesh_elm_idx);
         if ( it != element_to_map_.end() ) return element_starts_[it->second];
         else return ElementCacheMap::undef_elem_idx;
     }
 
     /// Return end position of element chunk in FieldValueCache
     inline unsigned int element_chunk_end(unsigned int mesh_elm_idx) const {
-        std::map<unsigned int, unsigned int>::const_iterator it = element_to_map_.find(mesh_elm_idx);
+        std::unordered_map<unsigned int, unsigned int>::const_iterator it = element_to_map_.find(mesh_elm_idx);
         if ( it != element_to_map_.end() ) return element_starts_[it->second+1];
         else return ElementCacheMap::undef_elem_idx;
     }
 
+    /// Return begin position of element chunk in FieldValueCache
+    inline unsigned int element_chunk_begin_new(unsigned int elm_patch_idx) const {
```

Comment:
Remove "old" variants and possibly introduce `find_patch_element(mesh_elem_idx)`.